components/script/dom/htmlheadelement.rs

```diff
@@ -80,6 +81,56 @@ impl HTMLHeadElement {
             }
         }
     }
+
+    /// <https://w3c.github.io/webappsec-csp/#meta-element>
+    pub fn set_content_security_policy(&self) {
+        let doc = document_from_node(self);
+
+        if doc.GetHead().as_deref() != Some(self) {
+            return;
+        }
+
+        let mut csp_list: Option<CspList> = None;
+        let node = self.upcast::<Node>();
+        let candinates = node
+            .traverse_preorder(ShadowIncluding::No)
+            .filter_map(DomRoot::downcast::<Element>)
+            .filter(|elem| elem.is::<HTMLMetaElement>())
+            .filter(|elem| {
+                elem.get_string_attribute(&local_name!("http-equiv"))
+                    .to_ascii_lowercase() ==
+                    "content-security-policy".to_owned()
+            })
+            .filter(|elem| {
+                elem.get_attribute(&ns!(), &local_name!("content"))
+                    .is_some()
+            });
+
+        for meta in candinates {
+            if let Some(ref content) = meta.get_attribute(&ns!(), &local_name!("content")) {
+                let content = content.value();
+                let content_val = content.trim();
+                if !content_val.is_empty() {
+                    match csp_list {
+                        Some(ref mut csp_list) => csp_list.append(CspList::parse(
+                            content_val,
```

Comment:
Let's de-duplucate the CsoList::parse call by storing it in a local variable before the match.